### PR TITLE
Issue #14436: Enforced naming convention for `InnerAssignmentCheckTest`

### PR DESCRIPTION
**Pull request by abhishekmaity**
Part of #14436

### CODE REVIEW

**Review by MANISH-K-07 — COMMENTED** [review]
Please read the issue description and understand the requirement here.
Find examples of ideal updates in the issue description.

**Comment by nrmancuso** [comment]
@abhishekmaity I am also putting this PR into draft mode, I suggest to focus on one PR at a time for now. Let's get https://github.com/checkstyle/checkstyle/pull/14539 across the finish line first.

**Comment by abhishekmaity** [comment]
@nrmancuso @romani PR ready for review

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void testIt() throws Exception {
     }
 
     @Test
-    public void testMethod() throws Exception {
+    public void testInputInnerAssignmentMethod() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:52]: The file name [InputInnerAssignmentMethod.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignmentMethod.java) doesn't seem accurate.
Please look for examples of ideal updates in the issue description

I would recommend something like : `InputInnerAssignmentNoViolationIterativeStatements`.java

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public void testMethod() throws Exception {
     }
 
     @Test
-    public void testDemoBug1195047Comment3() throws Exception {
+    public void testInputInnerAssignmentCheckLoopsAndConditionals() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:61]: The test method name doesn't include the word `Input` by convention

Please change : 
testInputInnerAssignmentCheckLoopsAndConditionals() --> `testInnerAssignmentCheckLoopsAndConditionals()`

Update the same for all such occurences.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ protected String getPackageLocation() {
     }
 
     @Test
-    public void testIt() throws Exception {
+    public void testInputInnerAssignment() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:39]: The file name [InputInnerAssignment.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignment.java) doesn't really specify what the input file is actually testing.
You are required to analyse and rename such files according to test functionality as part of this issue....

I would recommend something like : `InputInnerAssignmentCheckLocalVariables`.java

**Review by MANISH-K-07 — COMMENTED** [review]
Please resolve conflicts with failing CI (https://circleci.com/gh/checkstyle/checkstyle/527154)
Looks like your PR consists of 4 commits. Squash and rebase to one

Items :

**Comment by abhishekmaity** [comment]
> Please resolve conflicts with failing CI (https://circleci.com/gh/checkstyle/checkstyle/527154) Looks like your PR consists of 4 commits. Squash and rebase to one
> 
> Items :

Suggestion implemented and all commits squashed

**Comment by abhishekmaity** [comment]
@nrmancuso @romani PR ready for review

**Comment by abhishekmaity** [comment]
1. **Cirrus -JDK** failed -> `Failed to start an instance: FAILED_PRECONDITION: Monthly compute limit exceeded!`
2. **ci/circleci: no-error-xwiki**  ->`checkstyle violation`, but code has passed `mvn install verify` with BUILD SUCCESS at local
3. **ci/circleci: run-inspections** -> showing failed because  `There are inspection problems`

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 4]
```diff
@@ -36,7 +36,7 @@ protected String getPackageLocation() {
     }
 
     @Test
-    public void testIt() throws Exception {
+    public void testInputInnerAssignmentCheckLocalVariables() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:39]: Not done :(
Please check my comment at https://github.com/checkstyle/checkstyle/pull/14543#discussion_r1502054425

test method names should not have "`Input`" by convention.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 5]
```diff
@@ -45,20 +45,20 @@ public void testIt() throws Exception {
             "73:19: " + getCheckMessage(MSG_KEY),
         };
         verifyWithInlineConfigParser(
-                getPath("InputInnerAssignment.java"), expected);
+                getPath("InputInnerAssignmentCheckLocalVariables.java"), expected);
     }
 
     @Test
-    public void testMethod() throws Exception {
+    public void testInputInnerAssignmentNoViolationIterativeStatements() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:52]: Same as above....

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 6]
```diff
@@ -74,19 +74,19 @@ public void testDemoBug1195047Comment3() throws Exception {
             "26:32: " + getCheckMessage(MSG_KEY),
         };
         verifyWithInlineConfigParser(
-                getPath("InputInnerAssignmentDemoBug1195047Comment3.java"), expected);
+                getPath("InputInnerAssignmentCheckLoopsAndConditionals.java"), expected);
     }
 
     @Test
-    public void testLambdaExpression() throws Exception {
+    public void testInputInnerAssignmentLambdaExpressions() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:81]: Same as above....

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java [hunk 7]
```diff
@@ -74,19 +74,19 @@ public void testDemoBug1195047Comment3() throws Exception {
...
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
         verifyWithInlineConfigParser(
                 getPath("InputInnerAssignmentLambdaExpressions.java"),
                 expected);
     }
 
     @Test
-    public void testInnerAssignmentNotInLoopContext() throws Exception {
+    public void testInputInnerAssignmentNotInLoopContext() throws Exception {
```
Comment (MANISH-K-07) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheckTest.java:89]: Same as above....

**Review by MANISH-K-07 — COMMENTED** [review]
@abhishekmaity , 
Looks like you have leaked unrelated changes from previously merged commits into this PR.

Please revert all unnecessary changes and rebase. The PR needs to address only one issue and has to be clean

**Comment by abhishekmaity** [comment]
closing due to leakages from unrelated changes

**Pull request closed without merging**